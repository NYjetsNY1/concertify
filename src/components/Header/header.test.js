import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Header from './Header';

describe('Header', () => {
  test('renders correct content', () => {
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {}, fetch: () => {} }}>
          <Header />
        </App>,
      )
      .toJSON();
    expect(wrapper.type).toBe('div');
    expect(wrapper.props.className).toBe('root');

    const inner_container = wrapper.children[0];
    expect(inner_container.type).toBe('div');
    expect(inner_container.props.className).toBe('container');

    const nav_area = inner_container.children[0];
    expect(nav_area.props.role).toBe('navigation');

    const brand_name = inner_container.children[1];
    expect(brand_name.type).toBe('a');
    expect(brand_name.props.className).toContain('brand');
    expect(brand_name.children[0]).toBe('Concertify');
  });
});
