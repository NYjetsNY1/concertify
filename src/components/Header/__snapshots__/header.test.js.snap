// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correct content 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <div
      className="root"
      role="navigation"
    >
      <a
        className="link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <span
        className="spacer"
      >
         | 
      </span>
      <a
        className="link"
        href="/auth"
      >
        Log in
      </a>
    </div>
    <a
      className="brand"
      href="/"
      onClick={[Function]}
    >
      Concertify
    </a>
  </div>
</div>
`;
